### PR TITLE
Creates a base-application.gradle 

### PR DESCRIPTION
**Pull request by vegaro**
Created a `base-application.gradle` to remove duplicated configuration in the applications we have in the repository.

Suggested by @NachoSoto in https://github.com/RevenueCat/purchases-android/pull/445#discussion_r785022898

### CODE REVIEW

**Review by NachoSoto — APPROVED** [review]
Yay 🥳

File: examples/MagicWeather/app/build.gradle [hunk 1]
```diff
@@ -1,20 +1,12 @@
...
-}
+apply from: "$rootProject.projectDir/base-application.gradle"
 
 android {
-    compileSdkVersion 29
-    buildToolsVersion "30.0.3"
-
     defaultConfig {
         applicationId "com.revenuecat.purchases_sample"
         minSdkVersion 26
```
Comment (beylmk) [inline examples/MagicWeather/app/build.gradle:6]: is this a diff minSdk for a reason?

File: examples/MagicWeather/app/build.gradle [hunk 2]
```diff
@@ -1,20 +1,12 @@
...
-        targetSdkVersion 29
+        targetSdkVersion compileVersion
```
Comment (beylmk) [inline examples/MagicWeather/app/build.gradle:7]: we can just remove this now right?

**Review by beylmk — APPROVED** [review]
🥳  this is AWESOME. minor comments but they could be a follow-up PR

File: examples/MagicWeather/app/build.gradle (hunk 1 above)
Comment (vegaro) [reply examples/MagicWeather/app/build.gradle:6]: I don't know, I think it has a dependency that requires 26, but I am not sure

**Pull request merged**